interfaces/builtin/snapd_control.go

```diff
@@ -35,22 +41,38 @@ const snapdControlBaseDeclarationSlots = `
     deny-auto-connection: true
 `
 
-// http://bazaar.launchpad.net/~ubuntu-security/ubuntu-core-security/trunk/view/head:/data/apparmor/policygroups/ubuntu-core/16.04/snapd-control
 const snapdControlConnectedPlugAppArmor = `
 # Description: Can manage snaps via snapd.
 
 /run/snapd.socket rw,
 `
 
+type snapControlInterface struct {
+	commonInterface
+}
+
+func (iface *snapControlInterface) SanitizePlug(plug *interfaces.Plug) error {
+	refreshSchedule, ok := plug.Attrs["refresh-schedule"].(string)
+	if ok {
+		if refreshSchedule != "" && refreshSchedule != "managed" {
+			return fmt.Errorf("unsupported refresh-schedule value: %q", refreshSchedule)
+		}
+	}
+
+	return nil
+}
+
 func init() {
-	registerIface(&commonInterface{
-		name:                  "snapd-control",
-		summary:               snapdControlSummary,
-		implicitOnCore:        true,
-		implicitOnClassic:     true,
-		baseDeclarationPlugs:  snapdControlBaseDeclarationPlugs,
-		baseDeclarationSlots:  snapdControlBaseDeclarationSlots,
-		connectedPlugAppArmor: snapdControlConnectedPlugAppArmor,
-		reservedForOS:         true,
+	registerIface(&snapControlInterface{
```

Comment:
Can you please reformat so that this fits on one line:

```
registerIface(&snapControlInterface{commonInterface{
```

Then the rest should have smaller delta.